### DISCUSSION TITLE
Why these desktop containers?

### DISCUSSION THREAD

**Discussion by accetto** [accetto/ubuntu-vnc-xfce-g3#13] [Q&A] [answered]
Why do you develop these desktop containers? Isn't it againts the Docker's Zen `one container per process`?

**Comment by accetto** [comment, marked as answer]
My desktop Docker containers are intended to be used as encapsulated application environments for experimenting, testing and development.

Originally I've developed them as a replacement for the `VirtualBox` virtual machines on Windows, because the Microsoft's `Hyper-V` has been in conflict with the `VirtualBox` at the time.

Because I've found these containers very helpful and I use them daily on both Windows and Linux, I continue to developt them further.

I see the `Docker` as just an another tool in our toolkit. And as with all other tools - if it solves the problem, then just use it. :)

**Reply by rsettlage** [reply]
We use them in a similar fashion.  Get a GUI/interactive app up in going in one of these encapsulated environments.  Makes it easier to have it all in one container.

**Reply by accetto** [reply]
Thanks! :)

**Reply by rsettlage** [reply]
I should also mention that in another project, I combine 8 containers to a) make it easier to run a pipeline using components of each container and b) the combined size of the 8 containers is 6 GB where a single combined container is 1.2 GB total.  In this case, I have to pay for storage so the saved space, even if only temporarily duplicated is a big deal.  Docker Zen may not be as complete a picture as it is made out to be.
